### ISSUE TITLE
Add label smoothing to CopyNetSeq2Seq

### ISSUE THREAD

**Issue by JohnGiorgi** [allenai/allennlp#5276]
**Is your feature request related to a problem? Please describe.**

I am wondering if it is possible to add label smoothing to [`CopyNetSeq2Seq`](https://github.com/allenai/allennlp-models/blob/main/allennlp_models/generation/models/copynet_seq2seq.py). Label smoothing is implemented for the other allennlp-models under the [`generation`](https://github.com/allenai/allennlp-models/tree/main/allennlp_models/generation/models) module via [`sequence_cross_entropy_with_logits`](https://github.com/allenai/allennlp/blob/cf113d705b9054d329c67cf9bb29cbc3f191015d/allennlp/nn/util.py#L706).

**Describe the solution you'd like**

I think ideally, `CopyNetSeq2Seq` would be updated to use `sequence_cross_entropy_with_logits`, which would enable label smoothing in addition to some other features (like `alpha` and `gamma`). 

**Describe alternatives you've considered**

An alternative solution would be to skip `sequence_cross_entropy_with_logits` and just add label smoothing directly to `CopyNetSeq2Seq`, with a new parameter: `label_smoothing`.

**Additional context**

I've looked through the source code of `CopyNetSeq2Seq`, but I can't quite figure out how to implement either solution, or if they are even feasible given the complications that the copy mechanism introduces.

I would be happy to take a crack at this but I think I might need more guidance (if it's feasible and if so where to start).

**Comment by AkshitaB** [comment]
@JohnGiorgi Thanks for the suggestion. I'll be happy to discuss the details in the contribution. I think the simplest thing would be to add label smoothing to`CopyNetSeq2Seq` directly. Since the scores for generation and copying are computed separately, I think the best place to add this would be for the generation part. Let me know if that makes sense.

**Comment by JohnGiorgi** [comment]
Cool! Opened a PR here: https://github.com/allenai/allennlp-models/pull/287

**Comment by AkshitaB** [comment]
The PR has been merged. Closing.
